apps/src/dance/ai/utils/chooseEffects.ts

```diff
@@ -1,19 +1,29 @@
 import {cachedWeightsMappings} from '../constants';
-import {FieldKey, GeneratedEffect, EffectsQuality} from '../types';
-
+import {
+  FieldKey,
+  GeneratedEffect,
+  EffectsQuality,
+  CachedWeightsMapping,
+} from '../types';
 import {calculateOutputSummedWeights} from './calculateOutputSummedWeights';
-import {getSortedOptions} from './getSortedOptions';
 
 const NUM_RANDOM_TOP_OPTIONS = 3;
 const NUM_RANDOM_BOTTOM_OPTIONS = 20;
 
+type CalculateSummedWeights = (
+  emojis: string[],
+  weightMapping: CachedWeightsMapping
+) => [number, string][];
+
 /**
  * Chooses a random background effect, background color, and foreground effect associated
  * with the emojis the user selected.  If quality is GOOD, then choose one of the best
  * associated; if quality is BAD, then choose one of the worst associated.  Also return
  * the score for each returned value.
  * @param {array} selectedEmojis: the list of three emojis the user selected
  * @param {EffectsQuality} quality: whether we want good or bad scoring effects
+ * @param {CalculateSummedWeights} calculateSummedWeights: function to calculate weights (used for testing)
```

Comment:
Just curious: Is it standard to include 'used for testing' for a function parameter?

Reply:
Not sure, I just searched "for testing" in our repo and I often see us noting when we are exporting a function/etc explicitly for testing, or doing other behavior specifically for testing purposes. Less common to see in JSDocs, although I see at least [one instance here](https://github.com/code-dot-org/code-dot-org/blob/54a55a398c3508cab5e4e55da1a47f7471cc5d7a/apps/src/code-studio/initApp/project.js#L1651). Thought it was worth clarifying that our actual application doesn't use these params.

Reply:
Gotcha - that makes sense and it is useful to know!

Reply:
Maybe add 'used only for testing'?

Reply:
Good idea, that is much clearer! I'll update